### ISSUE TITLE
[PATCH] Maildir folder and courier Maildir++ support (quota)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7608]
|<img alt="sirius's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| sirius reported|
|-|-|
|Trac ID|trac#396|
|Type|enhancement|
|Created|2003-11-16 07:32:19Z|

Attachments:

* [twisted-1.1.0-maildir++.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2bc/2bc4a9de212381b2bbb88945fcc28f65c338b37a/5e3281cf6c8771582fc96d2939fb85ae3d9c9fff.diff) (104076 bytes) - added by sirius on 2003-11-16 07:32:19Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__396 396
type__enhancement enhancement
reporter__sirius sirius
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__mail mail
keywords__ 
time__1068967939000000 1068967939000000
changetime__1469247032854464 1469247032854464
version__ 
owner__ 
cc__exarkun cc__sirius
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="sirius's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|sirius commented|
|-|-|

```
#!html
<pre>
I have extended the twisted maildir implementation with folder support and the
courier maildir++ extension(quota). All maildir functionality is now
concentrated within a single (new) class called Maildir. The included patch also
updates the testcase for maildir support. I have tried to keep things as
backwards compatible as possible. The patch was developed for twisted-1.1.0, but
it currently also applies without errors to the CVS version.

I'm new to the Twisted project, so I would be very happy if you guys could give
some comment on this piece of code. I'm still testing it and I'm about to use it
in the project I'm actually working on.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
exarkun's probably the person this is most relevant to.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
I reviewed this patch shortly after it was submitted, but did not finish and
couldn't decide on the proper response.

I think that most of the functionality this patch provides is useful, but I'm
not certain how it is useful to Twisted.  I now think the best way to proceed
would be to provide this Maildir implementation as a standalone module that
implements a well defined interface, and then, if necessary, patch Twisted to
support objects of that interface.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

Doing that, then.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub commented|
|-|-|

In changeset fc120b8a9313fa73bbe70c9dc58442f22f44e696
```
#!CommitTicketReference repository="" revision="fc120b8a9313fa73bbe70c9dc58442f22f44e696"
Merge pull request #396 from twisted/8648-markrwilliams-replace-runcommand-with-check_output

Author: markrwilliams
Reviewer: rodrigc
Fixes: #8648

twisted.python._release.runCommand now uses check_output
```
